src/traces/scatter/calc.js

```diff
@@ -115,6 +115,10 @@ module.exports = function calc(gd, trace) {
     for(i = 0; i < serieslen; i++) {
         cd[i] = (isNumeric(x[i]) && isNumeric(y[i])) ?
             {x: x[i], y: y[i]} : {x: false, y: false};
+
+        if(trace.identifier && trace.identifier[i] !== undefined) {
```

Comment:
Okay, that makes sense. Only fear: people with pass `'7'` and `7`, and they'll be confused by the silent casting.